homeassistant/components/lamarzocco/coordinator.py

```diff
@@ -98,3 +147,15 @@ async def _async_handle_request(
         except RequestNotSuccessful as ex:
             _LOGGER.debug(ex, exc_info=True)
             raise UpdateFailed("Querying API failed. Error: %s" % ex) from ex
+
+    def async_get_ble_device(self) -> BLEDevice | None:
+        """Get a Bleak Client for the machine."""
+        # according to HA best practices, we should not reuse the same client
+        # get a new BLE device from hass and init a new Bleak Client with it
```

Comment:
Its ok to pass the address instead of the BLEDevice now as the wrapper will look it up and convert it for you now.

https://developers.home-assistant.io/docs/bluetooth/#best-practices-for-integration-authors

Reply:
But I can pass the BLE Device for now? What exactly qualifies as a connection though? Every time I contact the device or every time connected has switched to False once?

Reply:
> But I can pass the BLE Device for now?

You can pass address or BLEDevice. It doesn't matter anymore. If you pass BLEDevice, the backend will extract the address from it and exchange it for a BLEDevice object that from the scanner that is nearest the device that hasn't recently failed.  If you pass address it will do the same but is slightly faster (not in any significant way)

> What exactly qualifies as a connection though?

Calling the `.connect()` method

Reply:
I'll change the library and adapt this in a follow up PR for simplicity